scripts/migrate_from_s3_to_ssm.py

```diff
@@ -0,0 +1,158 @@
+#!/usr/bin/env python
+import sys
+import json
+from io import BytesIO
+import boto3
+from botocore.exceptions import ClientError
+
+FILE_NAME = 'ssm_ready_secrets.json'
+
+
+def decrypt_secrets(ciphertext, region):
+    """Decrypts the given ciphertext using AWS KMS
+
+    Args:
+        ciphertext (str): The raw, encrypted data to be decrypted
+        region (str): AWS region
+
+    Returns:
+        string: The decrypted plaintext
+
+    Raises:
+        ClientError
+    """
+    client = boto3.client('kms', region_name=region)
+    try:
+        response = client.decrypt(CiphertextBlob=ciphertext)
+    except ClientError:
+        print('An error occurred during KMS decryption')
+        raise
+    else:
+        return response['Plaintext']
+
+
+def pull_from_s3(bucket, key):
+    """Pull file contents from S3 into memory
+
+    Args:
+        bucket (boto3.resources.factory.s3.Bucket): The S3 Bucket resource
+        key (str): The key to pull
+
+    Returns:
+        None: An error occured
+        ciphertext: The encrypted contents of the file
+
+    Raises:
+        ClientError
+    """
+    io = BytesIO()
+
+    try:
+        bucket.download_fileobj(key, io)
+    except ClientError:
+        print(f"Error fetching {key} from s3")
+        raise
+    else:
+        return io.getvalue()
+
+
+def get_output_secrets(bucket, key, region):
+    """Pull the secrets from s3
+
+    Args:
+        bucket (boto3.resources.factory.s3.Bucket): The S3 Bucket resource
+        key (str): The key to download and decrypt
+    Returns:
+        None: An error occured
+        dict: The secrets from s3
+    """
+    try:
+        encrypted_secrets = pull_from_s3(bucket, key)
+        secrets = decrypt_secrets(encrypted_secrets, region)
+    except ClientError:
+        pass
+    else:
+        return json.loads(secrets)
+
+
+def save_to_fs(secrets):
+    """Save the secrets to the local FileSystem
+
+    Args:
+        secrets (dict): Secrets ready to be taken by set-from-file subcommand
+    """
+
+    saved = False
+
+    try:
+        with open(FILE_NAME, "w") as _fp:
+            json.dump(secrets, _fp)
+    except Exception:
+        print("Error saving secrets to FileSystem")
+    else:
+        print(f"Secrets saved to {FILE_NAME}")
+        saved = True
+
+    return saved
+
+
+if __name__ == "__main__":
+    import os
```

Comment:
Is there any particular reason the `imports` are here instead of at the top of the file?

Reply:
Just cleanliness and the fact these are only relevant if name... Since I have to append streamalert to path in order to import it due to the location of the script